tests/functionaltests/Tests/UIKitTests/UIButtonTests.mm

```diff
@@ -51,75 +36,98 @@
     ASSERT_TRUE([backingElement isKindOfClass:[WXFrameworkElement class]]);
 }
 
-TEST(UIButton, BackgroundColorChanged) {
-    __block UIButtonViewController* buttonVC;
+TEST(UIButton, TitleColorChanged) {
+    __block BOOL signaled = NO;
     __block NSCondition* condition = [[[NSCondition alloc] init] autorelease];
-    __block WXUIElement* backingElement = nil;
-
-    dispatch_sync(dispatch_get_main_queue(), ^{
-        buttonVC = [[UIButtonViewController alloc] init];
-
-        // TODO: Remove this line once we hook up to the root view controller which will trigger the view method
-        [buttonVC view];
 
-        // Extract the UIButton example from an existing XAMLCatalog cell in a TableViewController
-        NSIndexPath* indexPath = [NSIndexPath indexPathForRow:1 inSection:0];
-        UITableViewCell* cell = [buttonVC tableView : buttonVC.tableView cellForRowAtIndexPath:indexPath];
-        ASSERT_TRUE(cell);
+    UIButtonWithControlsViewController* buttonVC = [[[UIButtonWithControlsViewController alloc] init] autorelease];
+    ViewControllerHelper testHelper(buttonVC);
 
-        // TODO: Formalize a way to extract existing buttons and their states from our test or sample apps
-        UILabel* cellLabel = [cell.subviews objectAtIndex:1];
-        UIButton* cellButton = [cell.subviews objectAtIndex:2];
-        LOG_INFO(@"Cell subview[1] text: %@", cellLabel.text);
-
-        // We have to artificially ref the element since the block needs to keep it around
-        backingElement = [cellButton xamlElement];
-        [backingElement retain];
-
-        // Register callback and wait for the property changed event to trigger
-        int64_t callbackToken = [backingElement
-            registerPropertyChangedCallback:[WXCControl backgroundProperty]
-                            callback:^(WXDependencyObject* sender, WXDependencyProperty* dp) {
+    dispatch_sync(dispatch_get_main_queue(), ^{
+        // Extract UIButton.titleLabel control to verify its visual state
+        WXFrameworkElement* titleElement = [buttonVC.button.titleLabel xamlElement];
+        ASSERT_TRUE(titleElement);
```

Comment:
I had to retract this change - we have to investigate if the _MSG version is or _OBJNE is compatible. I'll file a bug to track. 

---
In reply to: [97399064](https://github.com/Microsoft/WinObjC/pull/1771#discussion_r97399064) [](ancestors = 97399064)